app/controllers/provider_interface/offers_controller.rb

```diff
@@ -52,13 +52,16 @@ def show
     def update
       @wizard = OfferWizard.new(offer_store)
       if @wizard.valid?(:save)
-        ChangeOffer.new(actor: current_provider_user,
-                        application_choice: @application_choice,
-                        course_option: @wizard.course_option,
-                        update_conditions_service: update_conditions_service).save!
-        @wizard.clear_state!
-
-        flash[:success] = t('.success')
+        begin
+          ChangeOffer.new(actor: current_provider_user,
+                          application_choice: @application_choice,
+                          course_option: @wizard.course_option,
+                          update_conditions_service: update_conditions_service).save!
+          @wizard.clear_state!
+          flash[:success] = t('.success')
+        rescue IdenticalOfferError
+          flash[:success] = t('.success')
```

Comment:
Would we still want to do a `@wizard.clear_state`, just to make sure things aren't left hanging?

Reply:
Yes, good point! Updated.